clusterctl/clusterdeployer/clusterdeployer.go

```diff
@@ -127,37 +138,46 @@ func (d *ClusterDeployer) Create(cluster *clusterv1.Cluster, machines []*cluster
 	}
 	defer closeClient(bootstrapClient, "bootstrap")
 
+	if cluster.Namespace == "" {
+		cluster.Namespace = bootstrapClient.GetContextNamespace()
+	}
+
+	err = bootstrapClient.EnsureNamespace(cluster.Namespace)
+	if err != nil {
+		return fmt.Errorf("unable to ensure namespace %q in bootstrap cluster: %v", cluster.Namespace, err)
+	}
+
 	glog.Info("Applying Cluster API stack to bootstrap cluster")
-	if err := d.applyClusterAPIStack(bootstrapClient); err != nil {
+	if err := d.applyClusterAPIStack(bootstrapClient, cluster.Namespace); err != nil {
 		return fmt.Errorf("unable to apply cluster api stack to bootstrap cluster: %v", err)
 	}
 
 	glog.Info("Provisioning target cluster via bootstrap cluster")
 
-	glog.Infof("Creating cluster object %v on bootstrap cluster", cluster.Name)
+	glog.Infof("Creating cluster object %v on bootstrap cluster in namespace %v", cluster.Name, cluster.Namespace)
```

Comment:
all the other places above that namespace is printed use %q instead of %v. Is there a reason that this file does it in a different way?